src/dll/edbgEcmdDll.C

```diff
@@ -216,108 +162,59 @@ static uint32_t findChipUnitType(const ecmdChipTarget &i_target, uint64_t i_addr
-
-  if (cuString == "c")          o_P9CU = PU_C_CHIPUNIT;
-  else if (cuString == "eq")    o_P9CU = PU_EQ_CHIPUNIT;
-  else if (cuString == "ex")    o_P9CU = PU_EX_CHIPUNIT;
-  else if (cuString == "xbus")  o_P9CU = PU_XBUS_CHIPUNIT;
-  else if (cuString == "obus")  o_P9CU = PU_OBUS_CHIPUNIT;
-  else if (cuString == "nv")    o_P9CU = PU_NV_CHIPUNIT;
-  else if (cuString == "pec")   o_P9CU = PU_PEC_CHIPUNIT;
-  else if (cuString == "phb")   o_P9CU = PU_PHB_CHIPUNIT;
-  else if (cuString == "mi")    o_P9CU = PU_MI_CHIPUNIT;
-  else if (cuString == "dmi")   o_P9CU = PU_DMI_CHIPUNIT;
-  else if (cuString == "mcs")   o_P9CU = PU_MCS_CHIPUNIT;
-  else if (cuString == "mca")   o_P9CU = PU_MCA_CHIPUNIT;
-  else if (cuString == "mcbist")  o_P9CU = PU_MCBIST_CHIPUNIT;
-  else if (cuString == "perv")  o_P9CU = PU_PERV_CHIPUNIT;
-  else if (cuString == "ppe")   o_P9CU = PU_PPE_CHIPUNIT;
-  else if (cuString == "sbe")   o_P9CU = PU_SBE_CHIPUNIT;
-  else if (cuString == "capp")  o_P9CU = PU_CAPP_CHIPUNIT;
-  else if (cuString == "mc")    o_P9CU = PU_MC_CHIPUNIT;
-  else {
-    return out.error(EDBG_GENERAL_ERROR, FUNCNAME, "Unknown chip unit:%S\n", cuString.c_str());
-  }
+/**
+  * @brief Get chip type
+  *
+  * @param std::string o_chipType - Chip type as output
+  *
+  * @return Upon success, ECMD_SUCCESS will be returned.  A reason code will
+  *         be returned if the execution fails.
+  */
+static uint32_t getChipType(std::string &o_chipType) {
+    int rc=ECMD_SUCCESS;
+    uint32_t l_chip_id = 0;
+    uint32_t l_address = 0x100a; //Bits 12:19 tells chipID
+    ecmdDataBuffer l_data;
+    ecmdChipTarget l_target;
+
+    l_target.cageState =  l_target.nodeState = ECMD_TARGET_FIELD_VALID;
+    l_target.slotState = l_target.posState = ECMD_TARGET_FIELD_VALID;
+
+    //Read the 0x100a cfam to get the chip id
+    rc = dllGetCfamRegister(l_target, l_address, l_data);
+    if (rc){
+        return out.error(EDBG_READ_ERROR, FUNCNAME, "cfam read failure!\n");
+    }
 
-  return rc;
+    //bit 12:19 indicates the chip id and based on that unique id we determine
+    //the chip type.
+    l_chip_id = (l_data.getWord(0) & 0x000ff000) >> 12;
+    switch(l_chip_id) {
+        //0x223'd1'049 - p9
+        case 0xd1:
+            o_chipType = CHIP_TYPE_P9;
+            break;
+
+        default:
+            return out.error(EDBG_GENERAL_ERROR,FUNCNAME, "Unrecognised Chip type" PRIx32 "\n", l_chip_id);
+    }
+    return rc;
 }
 
 // Load the device tree and initialise the targets
 static int initTargets(void) {
-  int fd;
-  void *fdt;
-  struct stat stat;
-  static int done = 0;
-
-  if (!done) {
-    done = 1;
-
-    // The user sets their device tree via this variable. If not set, fail
-    // Would be nice to also set via --device on the cmdline, but currently
-    // there is an order of operations problem.
-    // Longer term libpdbg will be able to auto-detect the correct thing to use.
-    char * devTree = getenv("EDBG_DTB");
```

Comment:
The EDBG_DTB variable can't just be removed.  There are places that set this variable that would break without it.  If pdbg_targets_init doesn't require the user to open the dtb any more, that's great - but PDBG_DTB would need to be set based on EDBG_DTB if it's set.

Reply:
Currently, pdbg itself will take care of using its own internal version of the device tree if not set explicitly.  so, EDBG doesn't have to keep its own version of the device tree copy. I have tested this on p9 system and it is working good

Reply:
You have missed the use case where EDBG_DTB=none.  That is used by the code to not even do the pdbg init because not all features of the code use pdbg, like VPD update.  If you want to consolidate down to just PDBG_DTB, that is fine - but you need to update the other places in the repo that set EDBG_DTB and preserve the none use case.

Reply:
I did not see any reference to EDBG_DTB anywhere else in the code. If you can let me know were and how to test vpd i can verify. If you want to keep both the codes.  If the VPD functions are not using the devtree or the current pdbg_targets_init() function then i would feel the current change is safe

Reply:
If you grep the repo EDBG_DTB is set in command scripts and the like that call into edbg.  If you preserve the 'none' functionality, that should be fine - it doesn't have to be directly tested.  That would be preserved preferably through replacing the EDBG_DTB reference with PDBG_DTB and then preserving a PDBG_DTB=none code path that doesn't call pdbg init.

Reply:
Yes, Thats preserved

Reply:
Just realized. we didn't touch the original edbgWrapper.sh and we kept as it is and all the modifications done w.r.t to bmc is in  edbgBmcWrapper.sh

Reply:
root@tacoma1z-bigcorebmc1:/usr/bin# export EDBG_DTB=none
root@tacoma1z-bigcorebmc1:/usr/bin# export EDBG_HOME=/usr/
root@tacoma1z-bigcorebmc1:/usr/bin# export ECMD_EXE=/usr/bin/edbg 

root@tacoma1z-bigcorebmc1:/usr/bin# env
SHELL=/bin/sh
EDITOR=vi
PWD=/usr/bin
LOGNAME=root
HOME=/home/root
ECMD_EXE=/usr/bin/edbg
SSH_CONNECTION=::ffff:9.3.62.216 32924 ::ffff:9.3.86.184 22
TERM=xterm
USER=root
SHLVL=1
SYSTEMD_PAGER=
EDBG_HOME=/usr/
PS1=\u@\h:\w\$ 
SSH_CLIENT=::ffff:9.3.62.216 32924 22
PATH=/usr/local/bin:/usr/bin:/bin:/usr/local/sbin:/usr/sbin:/sbin
EDBG_DTB=none
SSH_TTY=/dev/pts/3
OLDPWD=/usr/bin/bkup
_=/usr/bin/env

root@tacoma1z-bigcorebmc1:/usr/bin# getscom pu 50009 -all
pu      k0:n0:s0:p00       0x8540508300000000
pu      k0:n0:s0:p01       0x8436020300000000
/usr/bin/edbg getscom pu 50009 -all